hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py

```diff
@@ -373,8 +373,15 @@ def cancel(self, trading_pair: str, client_order_id: str):
         safe_ensure_future(self.execute_cancel(trading_pair, client_order_id))
         return client_order_id
 
-    async def execute_cancel(self, trading_pair: str, client_order_id: str):
+    async def execute_cancel(self, trading_pair: str, client_order_id: str) -> Dict[str, Any]:
         try:
+            # Checks if order is not being tracked or order is waiting for created confirmation.
+            tracked_order: Optional[BinancePerpetualsInFlightOrder] = self._in_flight_orders.get(client_order_id, None)
+            if not tracked_order or not tracked_order.exchange_order_id_update_event.is_set():
+                self.logger().error(f"Error cancelling order {client_order_id}. "
+                                    f"Order not being tracked or waiting order created confirmation...")
```

Comment:
Instead of checking if the event is set I think it would be better to check if the exchange_order_id is not None. Also maybe it would be more declarative to have a `is_local()` function in the inflight order returning true when the order does not have the external id